### PR TITLE
plugdata: init at 0.8.0

### PR DESCRIPTION
**Pull request by PowerUser64**
## Description of changes

This PR adds [plugdata](https://plugdata.org/), a wrapper around Pure Data that enables running it as a plugin.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by PowerUser64** [comment]
Plug Data is going to be receiving a big update to v0.8.0 in the next two weeks or so, so might be best to wait until then to merge. However, review is still appreciated, as the only change to get the new version is to the commit hash and version number.

**Comment by PowerUser64** [comment]
TODO: add icon to desktop entry

**Comment by PowerUser64** [comment]
Updated to v0.8.0. Desktop entry is created but the icon isn't applied yet.

**Comment by PowerUser64** [comment]
Desktop icon now displays, previously unrealized issues with x11 libraries not being included are now solved too.

**Comment by PowerUser64** [comment]
I would like to build the latest version of plugdata (v0.8.2), but I get a ton of linker errors when I do. Any suggestions?

File: pkgs/applications/audio/plugdata/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/plugdata/default.nix:37]:
```suggestion
    repo = "plugdata";
```
This is always this value, no matter if we overwrite pname

File: pkgs/applications/audio/plugdata/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/plugdata/default.nix:47]: You are sure imagemagick isn't enough?

File: pkgs/applications/audio/plugdata/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/plugdata/default.nix:72]: And that breaks it?

File: pkgs/applications/audio/plugdata/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/plugdata/default.nix:74]:
```suggestion
  postPatch = ''
```

File: pkgs/applications/audio/plugdata/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/plugdata/default.nix:75]: Please remove the set -x

File: pkgs/applications/audio/plugdata/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/plugdata/default.nix:83]: Please use relative directories. The top directoroy /build/ is not on all platforms

File: pkgs/applications/audio/plugdata/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/plugdata/default.nix:88]: Can we remove the commented line?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/plugdata/default.nix
Comment (PowerUser64) [reply pkgs/applications/audio/plugdata/default.nix:72]: Yeah, without these two things (`env.NIX_LDFLAGS` and `dontPatchELF`), a few things in the application won't work. The most noticeable one is dragging tabs outside of the main window (to move them to a new one) makes them invisible once they leave the boarder of the window.

image

Also, if the application has a titlebar (like when running as a plugin inside a DAW), it will think the cursor is offset from where it really is.

There are a few other things that happen, but these are the most notable. From what I've seen, these things apply to all applications that use the JUCE framework.

File: pkgs/applications/audio/plugdata/default.nix
Comment (PowerUser64) [reply pkgs/applications/audio/plugdata/default.nix:47]: I'm using it to generate resized icons. Might be able to switch it over, I'll see.

File: pkgs/applications/audio/plugdata/default.nix
Comment (PowerUser64) [reply pkgs/applications/audio/plugdata/default.nix:74]: What is the difference between these?

File: pkgs/applications/audio/plugdata/default.nix
Comment (PowerUser64) [reply pkgs/applications/audio/plugdata/default.nix:83]: ooh! good to know

File: pkgs/applications/audio/plugdata/default.nix
Comment (SuperSandro2000) [reply pkgs/applications/audio/plugdata/default.nix]: If you overwrite patchPhase then patches breaks.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 253056` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>plugdata</li>
  </ul>
</details>

**Pull request merged**
